tensorflow/core/kernels/data/experimental/auto_shard_dataset_op_test.cc

```diff
@@ -0,0 +1,279 @@
+          /*breakpoints*/ {0, 1}};
+}
+
+// Test Case 3: the number of outputs could not be evenly divided by
+// num_workers.
+TestCase TestCase3() {
+  return {/*range_data_param*/ {0, 10, 1},
+          /*num_workers*/
+          DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {4}),
+          /*index*/
+          DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {3}),
+          /*expected_outputs*/
+          {DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {3}),
+           DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {7})},
+          /*expected_output_dtypes*/ {DT_INT64},
+          /*expected_output_shapes*/ {PartialTensorShape({})},
+          /*expected_cardinality*/ 2,
+          /*breakpoints*/ {0, 1, 5}};
+}
+
+TestCase IndexGreaterNumWorkersCase() {
+  return {/*range_data_param*/ {0, 10, 1},
+          /*num_workers*/
+          DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {5}),
+          /*index*/
+          DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {7}),
+          /*expected_outputs*/ {},
+          /*expected_output_dtypes*/ {DT_INT64},
+          /*expected_output_shapes*/ {PartialTensorShape({})},
+          /*expected_cardinality*/ 0,
+          /*breakpoints*/ {}};
+}
+
+TestCase NegativeIndexTestCase() {
+  return {/*range_data_param*/ {0, 10, 1},
+          /*num_workers*/
+          DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {5}),
+          /*index*/
+          DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {-3}),
+          /*expected_outputs*/ {},
+          /*expected_output_dtypes*/ {DT_INT64},
+          /*expected_output_shapes*/ {PartialTensorShape({})},
+          /*expected_cardinality*/ 0,
+          /*breakpoints*/ {}};
+}
+
+TestCase NegativeNumWorkersTestCase() {
+  return {/*range_data_param*/ {0, 10, 1},
+          /*num_workers*/
+          DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {-3}),
+          /*index*/
+          DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {1}),
+          /*expected_outputs*/ {},
+          /*expected_output_dtypes*/ {DT_INT64},
+          /*expected_output_shapes*/ {PartialTensorShape({})},
+          /*expected_cardinality*/ 0,
+          /*breakpoints*/ {}};
+}
+
+TestCase ZeroNumWorkersTestCase() {
+  return {/*range_data_param*/ {0, 10, 1},
+          /*num_workers*/
+          DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {0}),
+          /*index*/
+          DatasetOpsTestBase::CreateTensor<int64>(TensorShape({}), {1}),
+          /*expected_outputs*/ {},
+          /*expected_output_dtypes*/ {DT_INT64},
+          /*expected_output_shapes*/ {PartialTensorShape({})},
+          /*expected_cardinality*/ 0,
+          /*breakpoints*/ {}};
+}
+
+class ParameterizedAutoShardDatasetOpTest
+    : public AutoShardDatasetOpTest,
+      public ::testing::WithParamInterface<TestCase> {};
+
+TEST_P(ParameterizedAutoShardDatasetOpTest, GetNext) {
```

Comment:
`AutoShardDatasetOp` returns `ShardDataset` which has been tested in [shard_dataset_op_test.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/data/shard_dataset_op_test.cc), so the other APIs are not tested here. Please feel free to let me know if more tests are needed here.